.github/workflows/CD.yml

```diff
@@ -523,41 +499,44 @@ jobs:
           key: "${{ env.PACKAGE_CACHE_KEY }}"
           path: build/packages
 
-      - name: Load Image
-        run: docker load < build/packages/$IMAGE_NAME.tar
+      - name: Login Dockerhub
+        uses: docker/login-action@v1
+        with:
+          username: "${{ secrets.DOCKERHUB_RELEASE_USERNAME }}"
+          password: "${{ secrets.DOCKERHUB_RELEASE_TOKEN }}"
 
-      - name: Configure AWS Credentials
-        uses: aws-actions/configure-aws-credentials@v1
+      - name: Login to Public Integration Test ECR
+        if: steps.cached_image.outputs.cache-hit != 'true'
+        id: login-ecr
+        uses: docker/login-action@v1
         with:
-          aws-access-key-id: ${{ secrets.RELEASE_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.RELEASE_SECRET }}
-          aws-region: us-west-2
+          registry: public.ecr.aws
+          username: ${{ secrets.RELEASE_KEY_ID }}
+          password: ${{ secrets.RELEASE_SECRET }}
+        env:
+          AWS_REGION: us-east-1
 
       - name: Compare version with Dockerhub latest
         id: version
         run: |
-          TAG=`cat build/packages/VERSION`
+          TAG="${{ needs.release-version-image.outputs.tag }}"
           TARGET_VERSION=$TAG bash tools/workflow/docker-version-compare.sh
 
-      - name: Tag latest and upload to ecr
+      - name: Pull image from integration test ECR, tag as latest and push to public release ECR and DockerHub
         if: steps.version.outputs.any-update == 'true'
-        run: |
-          aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/aws-observability
-          docker tag $IMAGE_NAME public.ecr.aws/$ECR_REPO:latest
-          docker push public.ecr.aws/$ECR_REPO:latest
-
-      - name: Login Dockerhub
-        uses: docker/login-action@v1
+        uses: akhilerm/tag-push-action@v2.0.0
         with:
-          username: "${{ secrets.DOCKERHUB_RELEASE_USERNAME }}"
-          password: "${{ secrets.DOCKERHUB_RELEASE_TOKEN }}"
+          src: public.ecr.aws/$ECR_REPO_INTEGRATION_TEST:${{ needs.release-version-image.outputs.testing_tag }}
```

Comment:
Since the image is based on condition ```if: steps.version.outputs.any-update == 'true'```. Therefore, if the condition is not true, we cannot find the re-tagged image though. Unless, if we want to rotate the steps ( push versioning image first then mark latest later) then we could do it though.

Reply:
Maybe I'm missing something, but isn't the image pushed to the release ECR as part of `release-version-image`?

Reply:
Yeah. I thought that you mentioned [about the image ](https://github.com/aws-observability/aws-otel-collector/pull/860/files#diff-a64cbb6e953e697cb4302ab31af9ebbd95a8de6216f874bb0ffdefce8e3c1cd9R531-R538) that is tagged as latest so answer the wrong one.